rules-reviewed/quarkus/springboot/springboot-security-to-quarkus.windup.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="springboot-security-to-quarkus"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset gives hints to migrate from SpringBoot Security to Quarkus spring-security extension
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="springboot" />
+        <targetTechnology id="quarkus" />
+    </metadata>
+    <rules>
+        <rule id="springboot-security-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework.security" artifactId="spring-security-core"/>
+            </when>
```

Comment:
```suggestion
            <when>
            <project>
                <artifact groupId="org.springframework.security" artifactId="spring-security-core"/>
                </project>
            </when>
```
If you build a test data jar from spring initializr with the security dependency it will do it using a dependency of the security starter artifact in the pom. It will build with the starter-core jar we're looking for here in the lib folder and this rule won't find that. You'd need a rule like the one I suggest here. This works fine but I don't think it's as realistic a test scenario as the one I've outlined.

Reply:
I'm not sure I follow you.
I've created a project with start.spring.io , including Security.
When you build the project , and later you unzip the jar, you can see that in the lib folder there's an artifact "spring-security-core", that it's the one we are checking.

Reply:
I've checked that an analysis finds this file in the lib folder of a jar which was the scenario I was concerned about, so I'm happy this point doesn't need to be addressed any further.